DataDog/datadog-operator
# Remove automatic copying of previous Agent/ClusterAgent unknown annotations/labels

### What does this PR do?

Remove automatic copying of previous Agent/ClusterAgent unknown annotations/labels.
New fields in DatadogAgent CRD allows to specify annotations/labels we should carry over.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Deploy Agent and ClusterAgent.
- Annotate both of them with random content, the added annotations should be removed with the next `.Spec` change (won't be removed until then).
- Verify that any annotation/labels containing `datadoghq.com` is still kept.
- Verify that setting the `KeepLabels/KeepAnnotations` allows the added annotations/labels to stay (glob syntax).
